### ISSUE TITLE
consider dropping fieldAlignment static check

### ISSUE THREAD

**Issue by egernst** [kata-containers/kata-containers#3604]
I see fieldAlignment static check as an (over?) optimization that sacrifices developer time and often readability of our code.

I'd like to get other community member feedback on this topic. If we do keep it around, I suggest we have reference to appropriate tooling for automatically resolving, at least.

**Comment by fengwang666** [comment]
+1. This type of static check error is very time-consuming to fix. Unless there is an auto-format tool available, I'm in favor of dropping it.

**Comment by liubin** [comment]
Had anyone used this tool: 

```
go install golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment@latest
```

Seems it can do the fix for `fieldalignment` warnings.

**Comment by jodh-intel** [comment]
Agreed, it is super annoying / time consuming to fix manually, but the tool @liubin has found seems to work, which is great news!

@liubin - how about adding that to the static checker so it shows the diff output as in your example?

**Comment by zvonkok** [comment]
Ok, saw this issue too late, my latest PR has a lot of fixes for the structs in Kata, took years to get this right with all the comments. #4492 One thing that this provided is that some code does not use field names to initialize the structs which led and can lead to errors in the future if structs are reordered or extended. Could we add this as a warning so people who are really interested can still take a look but do not enforce it? Coming from a performance background I can tell that alignment sometimes matters but not in the general case. 

We do not have some HPC code where we're counting the cache-misses and thinking about access patterns of each field in a struct, or deciding if we should do an array-of-structs or structs-of-arrays for better access or SIMD programming. 

Comments are not yet preserved, https://github.com/golang/go/issues/20744. 

@jodh-intel Yes that would be great to at least have some indication of what the  tool is planning to do, but the current implementation of `fieldalignment` can only point you to the beginning of the struct not what is going to do if "fixed".

> "Programmers waste enormous amounts of time thinking about or worrying about, the speed of noncritical parts of their programs, and these attempts at efficiency actually have a strong negative impact when debugging and maintenance are considered. We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. Yet we should not pass up our opportunities in that critical 3%."

**Comment by zvonkok** [comment]
@fidencio @jodh-intel @liubin @fengwang666 @egernst This is still pretty annoying.
